inference-engine/src/mkldnn_plugin/nodes/mkldnn_concat_node.cpp

```diff
@@ -408,64 +422,66 @@ void MKLDNNConcatNode::initOptimalPrimitiveDescriptor() {
     }
 
     auto config = selected_pd->getConfig();
-    if (isConfigDefined(config))
-        return;
-
-    for (size_t i = 0; i < config.outConfs.size(); i++) {
-        if (config.outConfs[i].desc->isDefined())
-            continue;
+    if (!isDynamicNode() && !isConfigDefined(config)) {
+        for (size_t i = 0; i < config.outConfs.size(); i++) {
+            if (config.outConfs[i].desc->isDefined())
+                continue;
 
-        int num = getChildEdgeAt(i)->getOutputNum();
-        if (num >= 0) {
-            auto childConf = getChildEdgeAt(i)->getChild()->getSelectedPrimitiveDescriptor()->getConfig().inConfs[num];
-            childConf.desc = childConf.desc->cloneWithNewPrecision(config.outConfs[i].desc->getPrecision());
+            int num = getChildEdgeAt(i)->getOutputNum();
+            if (num >= 0) {
+                auto childConf = getChildEdgeAt(i)->getChild()->getSelectedPrimitiveDescriptor()->getConfig().inConfs[num];
+                childConf.desc = childConf.desc->cloneWithNewPrecision(config.outConfs[i].desc->getPrecision());
 
-            if (getChildEdgeAt(i)->getChild()->getSelectedPrimitiveDescriptor()) {
-                if (!childConf.desc->isDefined() && childConf.inPlace >= 0)
-                    getChildEdgeAt(i)->getChild()->initOptimalPrimitiveDescriptor();
+                if (getChildEdgeAt(i)->getChild()->getSelectedPrimitiveDescriptor()) {
+                    if (!childConf.desc->isDefined() && childConf.inPlace >= 0)
+                        getChildEdgeAt(i)->getChild()->initOptimalPrimitiveDescriptor();
 
-                if (childConf.desc->isDefined() && childConf.desc->isCompatible(*config.outConfs[i].desc)) {
-                    config.outConfs[i].desc = childConf.desc;
-                    continue;
+                    if (childConf.desc->isDefined() && childConf.desc->isCompatible(*config.outConfs[i].desc)) {
```

Comment:
childConf is copied, after `getChildEdgeAt(i)->getChild()->initOptimalPrimitiveDescriptor();`, it should be copied again to reflect the possible changes, otherwise checks in 439 will use old config info.